lib/utils/ring_buffer.c

```diff
@@ -152,16 +112,17 @@ uint32_t ring_buf_peek(struct ring_buf *buf, uint8_t *data, uint32_t size)
 	uint32_t total_size = 0U;
 	int err;
 
-	size = MIN(size, ring_buf_size_get(buf));
-
 	do {
 		partial_size = ring_buf_get_claim(buf, &src, size);
+		if (partial_size == 0) {
```

Comment:
Style, since the test is now at the top of the loop, you could refactor this into a while via e.g.:

```
while((partial_size = ring_buf_get_claim(buf, &src, size)) != 0) {
   ...
}
```

IIRC someone (MISRA?) doesn't want us using assignment in while expressions though, so it might need a wrapper that takes the size by reference and returns a boolean?  Or not, I don't remember what the rules are supposed to be anymore.

Reply:
I'm sure MISRA is against assignments within condition tests. And the loop has 2 exit conditions: one at the top and one at the bottom.